SOSS/dms/soss_centroids.py

```diff
@@ -0,0 +1,377 @@
+    :param poly_order: Order of the polynomial to fit to the extracted trace positions.
+    :param halfwidth: the size of the window used when computing the derivatives.
+    :param mode: Which trace values to use. Can be 'maxedge', 'minedge', 'mean' or 'combined'.
+    :param verbose: If set True some diagnostic plots will be made.
+
+    :type image: array[float]
+    :type header: astropy.io.fits.Header
+    :type mask: array[bool]
+    :type poly_order: int
+    :type halfwidth: int
+    :type mode: str
+    :type verbose: bool
+
+    :returns: xtrace, ytrace, tracewidth, param - The x, y coordinates of trace as computed from the best fit polynomial
+    and the best-fit polynomial parameters.
+    :rtype: Tuple(array[float], array[float], array[float])
+    """
+
+    # If no mask was given use all pixels.
+    if mask is None:
+        mask = np.zeros_like(image, dtype='bool')
+
+    # Call the script that determines the dimensions of the stack.
+    dimx, dimy, xos, yos, xnative, ynative, padding, refpix_mask = \
+        get_image_dim(image, header=header, verbose=verbose)
+
+    # Replace masked pixel values with NaNs.
+    image_masked = np.where(mask | ~refpix_mask, np.nan, image)
+
+    # Use edge trigger to compute the edges and center of the trace.
+    ytrace_max, ytrace_min, ytrace_best, widths_best = edge_trigger(image_masked, halfwidth=halfwidth, yos=yos, verbose=verbose)
+
+    # Use different y-positions depending on the mode parameter.
+    if mode == 'maxedge':
+        ytrace = ytrace_max
+    elif mode == 'minedge':
+        ytrace = ytrace_min
+    elif mode == 'mean':
+        ytrace = (ytrace_min + ytrace_max)/2.
+    elif mode == 'combined':
+        ytrace = ytrace_best
+    else:
+        raise ValueError('Unknow mode: {}'.format(mode))
+
+    # Fit the y-positions with a polynomial and use the result as the true y-positions.
+    xtrace = np.arange(dimx)
+    mask = np.isfinite(ytrace)
+
+    # If no polynomial order was given return the raw measurements.
+    if poly_order is None:
+        param = []
+    else:
+        param = robust_polyfit(xtrace[mask], ytrace[mask], poly_order)
+        ytrace = np.polyval(param, xtrace)
+
+    # If verbose visualize the result.
+    if verbose is True:
+        _plot_centroid(image_masked, xtrace, ytrace)
+
+    return xtrace, ytrace, widths_best, param
+
+
+def test_uncontam_centroids_edgetrig():
+    """"""
+
+    filename = os.path.join(PATH, 'SOSS_ref_2D_profile_SUBSTRIP256.fits.gz')
+
+    image, header = fits.getdata(filename, ext=2, header=True)
+    xtrace, ytrace, width, param = get_uncontam_centroids_edgetrig(image, header=header, verbose=True)
+
+    filename = '/home/talens-irex/Downloads/stack_256_ng3_DMS.fits'
+
+    image = fits.getdata(filename)
+    xtrace, ytrace, width, param = get_uncontam_centroids_edgetrig(image, verbose=True)
+
+    return
+
+
+def main():
```

Comment:
doc string , again if temporay flag with # TODO  to remove later

Reply:
Sorry you mean on main()? I've removed the contents but I think it's presence is required for any multiprocessing. Regardless I like having it there.

Reply:
you can have it - just add a doc string describing what it does (from the functions within it it looked like you were just using it as a test - hence the "if temporary" flag it as such - if it isn't then don't flag it as temporary but still have a doc string